src/main/java/net/neoforged/neoforge/transfer/handlers/wrappers/SimpleResourceHandler.java

```diff
@@ -0,0 +1,261 @@
+     * <p>
+     * <h5>IMPORTANT:</h5> This does not control your handler's logic in any way. Returning false, will not inherently prevent something from calling extract or change the result of that call,
+     * so you will still need to handle those scenarios. This is to allow things like logistics (pipes, searches, etc.) to be able to infer what it can do with the handler
+     * before actually operating.
+     * <p>
+     * It is also advised to not use the result of this call in extract.
+     * <p>
+     * If your handler can change size dynamically, then it may be wise to return true for this unless you know for certain a particular index would never be extractable from.
+     *
+     * <p>
+     * <b>This is expected to not change result unless it is accompanied by a capability invalidation.</b> (hence the note about dynamic size erring on the side of caution)
+     *
+     * @param index The index to check. <strong>Must be Non-Negative</strong>
+     * @return True if any resource can be extracted at the specified index, false otherwise.
+     */
+    public boolean supportsExtraction(int index) {
+        return handler.supportsExtraction(index);
+    }
+
+    /**
+     * Checks if the handler allows extraction from at least one index, regardless of the state of the handler. Also meaning this value is non-dynamic.
+     * <h5>IMPORTANT:</h5> This does not control your handler's logic in any way. Returning false, will not inherently prevent something from calling extract or change the result of that call,
+     * so you will still need to handle those scenarios. This is to allow things like logistics (pipes, searches, etc.) to be able to infer what it can do with the handler
+     * before actually operating.
+     * <p>
+     * It is also advised to not use the result of this call in extract.
+     * <p>
+     * <b>This is expected to not change result unless it is accompanied by a capability invalidation.</b> (hence the note about dynamic size erring on the side of caution)
+     *
+     * @return True if any resource can be extracted, false otherwise.
+     */
+    public boolean supportsExtraction() {
+        return handler.supportsExtraction();
+    }
+
+    /**
+     * Inserts a given amount of the resource into the handler at the given index.
+     *
+     * @param index      The index to insert the resource into. <strong>Must be Non-Negative</strong>
+     * @param resource   The resource to insert.
+     * @param amount     The amount of the resource to insert. <strong>Must be Non-Negative</strong>
+     * @param actionType The action to take, whether to simulate the insertion or actually perform it.
+     * @return The amount of the resource that was (or would have been, if simulated) inserted. <strong>Must be Non-Negative</strong>
+     */
+    public int insert(int index, T resource, int amount, TransferAction actionType) {
+        try (Transaction transaction = TransactionManager.open(null)) {
```

Comment:
I believe this and the other methods in this class will have issues if someone tries to use this helper class when already in a transaction context. I think it makes most sense when creating this wrapper for them to just pass in the current transaction context if there is one